hoomd/md/compute.py

```diff
@@ -0,0 +1,344 @@
+    those particles when requested. All specified :py:class:`ThermodynamicQuantities` objects can be added to a logger
+    for logging during a simulation, see :py:class:`hoomd.logging.Logger` for more details.
+
+    Note:
+        For pairwise terms :math:`U_{ij}` and :math:`F_{ij} \\cdot r_{ij}`, HOOMD-blue assigns half of each term to a
+        per-particle quantity. When filter selects a subset of the particles in the system, :py:class:`ThermodynamicQuantities`
+        sums these per-particle quantities over only particles selected by the filter.
+
+    Examples::
+
+        f = filter.Type('A')
+        compute.ThermodynamicQuantities(filter=f)
+    """
+
+    def __init__(self, filter):
+        super().__init__(filter)
+
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            thermo_cls = _md.ComputeThermo
+        else:
+            thermo_cls = _md.ComputeThermoGPU
+        group = self._simulation.state.get_group(self._filter)
+        self._cpp_obj = thermo_cls(self._simulation.state._cpp_sys_def, group, "")
+        super()._attach()
+
+    @log
+    def kinetic_temperature(self):
+        """
+        :math:`kT`, instantaneous thermal energy of the group (in energy units).
+
+        Calculated as:
+
+          .. math::
+
+            kT = 2 \\cdot \\frac{K}{N_{\\mathrm{dof}}}
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.kinetic_temperature
+        else:
+            return None
+
+    @log
+    def pressure(self):
+        """
+        :math:`P`, instantaneous pressure of the group (in pressure units).
+
+        Calculated as:
+
+        .. math::
+
+            P = \\left(\\frac{2}{D} \\cdot K_{\\mathrm{trans}} + \\frac{1}{2} \\cdot W \\right) / V,
+
+        where :math:`D` is the dimensionality of the system, :math:`V` is the total volume of the simulation box (or area in 2D), and :math:`W` is calculated as:
+
+        .. math::
+
+            W = \\frac{1}{2} \\sum_{i} \\sum_{j \\ne i} \\vec{F}_{ij} \\cdot \\vec{r_{ij}} + \\sum_{k} \\vec{F}_{k} \\cdot \\vec{r_{k}},
+
+        where :math:`i` and :math:`j` are particle tags, :math:`\\vec{F}_{ij}` are pairwise forces between particles and :math:`\\vec{F}_k` are forces due to explicit constraints, implicit rigid body constraints, external walls, and fields.
+        """
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.pressure
+        else:
+            return None
+
+    @log(flag='sequence')
+    def pressure_tensor(self):
+        """
+        (:math:`P_{xx}`, :math:`P_{xy}`, :math:`P_{xz}`, :math:`P_{yy}`, :math:`P_{yz}`, :math:`P_{zz}`).
+
+        Instantaneous pressure tensor of the group (in pressure units), calculated as:
+
+          .. math::
+
+              P_{ij} = \\left[  \\sum_{k\\in[0..N)} m_k v_{k,i} v_{k,j} +
```

Comment:
This also needs to be updated with the outer particle sums over particles in the filter and the inner sum running over all particles.